assets/app/scripts/directives/logViewer.js

```diff
@@ -82,13 +97,68 @@ angular.module('openshiftConsole')
                 });
               }
             };
-            $win.scroll(onScroll);
 
 
-            var onResize = _.debounce(updateScrollLinks, 50);
+
+            var attachScrollEvents = function() {
+              if(window.innerWidth < BREAKPOINTS.screenSmMin) {
+                $win.on('scroll', onScroll);
+                $cachedScrollableNode.off('scroll', onScroll);
+              } else {
+                $win.off('scroll', onScroll);
+                $cachedScrollableNode.on('scroll', onScroll);
+              }
+            };
+
+
+            // the class .target-logger-node is needed to adjust some
+            // css when the target is not the window.
+            // TODO: resize event breaks the affix, even with this if/else.
+            // however, on first load of either mobile or non this works fine.
+            var affix = function() {
+              if(window.innerWidth < BREAKPOINTS.screenSmMin) {
+                $affixableNode
+                  .removeClass('target-logger-node')
+                  .affix({
+                    target:  window,
+                    offset: {
+                        top:  $scope.followAffixTop || 0, // 390,
+                        bottom: $scope.followAffixBottom || 0 // 90
+                    }
+                  });
+              } else {
+                $affixableNode
+                  .addClass('target-logger-node')
+                  .affix({
+                    target:  $cachedScrollableNode,
+                    offset: {
+                        top: $scope.followAffixTop || 0, // 390,
+                        bottom: $scope.followAffixBottom || 0 // 90
+                    }
+                  });
+              }
+            };
```

Comment:
Yup. It doesn't register the target update so that causes more weird positioning.

Also tried `.data('bs.affix').options.target` per [this SO article](http://stackoverflow.com/questions/23797241/resetting-changing-the-offset-of-bootstrap-affix/23799130#23799130).  Was optimistic with this one since the fix suggested is by one of the BS creators :smile: